File: assets/process_screenshots.swift
#!/usr/bin/swift

import Foundation
import AppKit

// Configuration
// We use exact pixel dimensions
let targetWidth: Int = 2560
let targetHeight: Int = 1600
let backgroundColor = NSColor.black

func processImage(at url: URL) {
    let filename = url.lastPathComponent
    guard filename.hasPrefix("Screen_Shot") && filename.hasSuffix(".png") else { return }
    
    print("Processing \(filename)...")
    
    guard let image = NSImage(contentsOf: url) else {
        print("  Error: Could not load image")
        return
    }
    
    // Get exact pixel size of source
    guard let rep = image.representations.first as? NSBitmapImageRep else {
        print("  Error: Source is not a bitmap")
        return
    }
    
    let pixelWidth = CGFloat(rep.pixelsWide)
    let pixelHeight = CGFloat(rep.pixelsHigh)
    
    // Calculate scale to fit 2560x1600 box
    let scaleW = CGFloat(targetWidth) / pixelWidth
    let scaleH = CGFloat(targetHeight) / pixelHeight
    let scale = min(scaleW, scaleH)
    
    let newW = pixelWidth * scale
    let newH = pixelHeight * scale
    
    // Create output bitmap with exact pixel dimensions
    guard let outputRep = NSBitmapImageRep(
        bitmapDataPlanes: nil,
        pixelsWide: targetWidth,
        pixelsHigh: targetHeight,
        bitsPerSample: 8,
        samplesPerPixel: 4,
        hasAlpha: true,
        isPlanar: false,
        colorSpaceName: .deviceRGB,
        bytesPerRow: 0,
        bitsPerPixel: 0
    ) else {
        print("  Error: Could not create output bitmap")
        return
    }
    
    // Setup drawing context
    NSGraphicsContext.saveGraphicsState()
    let context = NSGraphicsContext(bitmapImageRep: outputRep)
    NSGraphicsContext.current = context
    
    // Fill background
    backgroundColor.setFill()
    NSRect(x: 0, y: 0, width: CGFloat(targetWidth), height: CGFloat(targetHeight)).fill()
    
    // Draw centered
    let x = (CGFloat(targetWidth) - newW) / 2
    let y = (CGFloat(targetHeight) - newH) / 2
    let destRect = NSRect(x: x, y: y, width: newW, height: newH)
    
    // Draw the original image into the new context
    // We prefer drawing the representation directly to avoid point-scaling issues
    rep.draw(in: destRect, 
             from: NSRect(x: 0, y: 0, width: pixelWidth, height: pixelHeight), 
             operation: .sourceOver, 
             fraction: 1.0, 
             respectFlipped: false, 
             hints: nil)
    
    NSGraphicsContext.restoreGraphicsState()
    
    // Save
    let outputFilename = "AppStore_" + filename.replacingOccurrences(of: " ", with: "_")
    let outputUrl = url.deletingLastPathComponent().appendingPathComponent(outputFilename)
    
    if let pngData = outputRep.representation(using: .png, properties: [:]) {
        try? pngData.write(to: outputUrl)
        print("  Saved to \(outputFilename) (\(targetWidth)x\(targetHeight) px)")
    }
}

let fileManager = FileManager.default
let currentDir = URL(fileURLWithPath: fileManager.currentDirectoryPath)

do {
    let files = try fileManager.contentsOfDirectory(at: currentDir, includingPropertiesForKeys: nil)
    print("Found \(files.count) files. Processing screenshots...")
    
    for file in files {
        processImage(at: file)
    }
    print("\nDone! Images saved as AppStore_*.png")
} catch {
    print("Error: \(error)")
}

File: assets/process_screenshots.py
import os
import subprocess
import sys

# Configuration
# Mac App Store Screenshot size (16:10 aspect ratio)
TARGET_WIDTH = 2560  # Retina preferred
TARGET_HEIGHT = 1600
BG_COLOR = "000000"  # Black letterboxing

def get_image_dimensions(path):
    # Use sips to get dimensions
    try:
        out = subprocess.check_output(['sips', '-g', 'pixelWidth', '-g', 'pixelHeight', path])
        lines = out.decode('utf-8').strip().split('\n')
        w = int(lines[1].split(':')[1].strip())
        h = int(lines[2].split(':')[1].strip())
        return w, h
    except:
        return 0, 0

def process_image(path):
    filename = os.path.basename(path)
    if not filename.startswith("Screen_Shot") or not filename.endswith(".png"):
        return

    print(f"Processing {filename}...")
    
    # Get current dimensions
    w, h = get_image_dimensions(path)
    if w == 0: return

    # Calculate scale to fit
    scale_w = TARGET_WIDTH / w
    scale_h = TARGET_HEIGHT / h
    scale = min(scale_w, scale_h)
    
    new_w = int(w * scale)
    new_h = int(h * scale)
    
    # 1. Resize main image to fit box
    temp_resized = f"temp_{filename}"
    subprocess.call(['sips', '-z', str(new_h), str(new_w), path, '--out', temp_resized], stdout=subprocess.DEVNULL)
    
    # 2. Pad to target size
    output_filename = f"AppStore_{filename.replace(' ', '_')}"
    subprocess.call([
        'sips', 
        '--padTo', str(TARGET_HEIGHT), str(TARGET_WIDTH), 
        '--padColor', BG_COLOR, 
        temp_resized, 
        '--out', output_filename
    ], stdout=subprocess.DEVNULL)
    
    # Cleanup
    os.remove(temp_resized)
    print(f"Created {output_filename} ({TARGET_WIDTH}x{TARGET_HEIGHT})")

def main():
    files = [f for f in os.listdir('.') if f.endswith('.png') and f.startswith('Screen_Shot')]
    files.sort()
    
    if not files:
        print("No screenshots found starting with 'Screen_Shot'...")
        return

    print(f"Found {len(files)} screenshots. Converting to {TARGET_WIDTH}x{TARGET_HEIGHT}...")
    
    for f in files:
        process_image(f)
        
    print("\nDone! Upload 'AppStore_...' files to App Store Connect.")

if __name__ == "__main__":
    main()
